# **week1 683380591-6 เดโชชิต โตวินัส**

## **LAB 1.1 Environment Setup & First Script**

In [ ]:
print("Hello world")

Hello world


In [ ]:
print("Hello, Python Course!")

Hello, Python Course!


In [ ]:
print("Hello,"+" Python"+" Course!")

Hello, Python Course!


# **Lab 1.2: Working with Variables and Basic Data Types**
- Objective: Students will define variables of different data types, perform simple operations, and display results.

2. Declare and print variables:

In [ ]:
age = 19
GPA = 3.79
My_favorite_color = "purple"
student = True
print("My age : ",age," ",type(age))
print("My GPA ; ",GPA," ",type(GPA))
print("My favorite color : ",My_favorite_color," ",type(My_favorite_color))
print("Student : ",student," ",type(student))

My age :  19   <class 'int'>
My GPA ;  3.79   <class 'float'>
My favorite color :  purple   <class 'str'>
Student :  True   <class 'bool'>


3. Perform simple arithmetic:

In [ ]:
num1 = 23
num2 = 70
sum = num1 + num2
different = num1 - num2
product = num1*num2
division = num1/num2
print(f"num1 = {num1}\nnum2 = {num2}\nresult")
print(f"sum = {sum}\ndifferent = {different}\nproduct = {product}\ndivision = {division}")

num1 = 23
num2 = 70
result
sum = 93
different = -47
product = 1610
division = 0.32857142857142857


4. String concatenation and f-strings:

In [ ]:
first_name = "Dechochit"
last_name = "Towinas"
full_name = first_name+" "+last_name
print(f"Hello, {full_name}!")

Hello, Dechochit Towinas!


5.User input and type conversion:

In [ ]:
num = input("Your favorite number : ")
print("Type num ",type(num))
int_num = int(num)
result = int_num+5
print(f"result {num} + 5: {result} ",type(result))

Your favorite number : 56
Type num  <class 'str'>
result 56 + 5: 61  <class 'int'>


# **Activity 1.1: Class Discussion - Uses of Scripting**
Format: Open discussion, possibly in small groups first, then full class.
Prompt: "Based on the introduction, where do you think scripting with Python could be applied in real-world scenarios? Think about tasks you do daily or things you've seen in other courses."
Expected outcomes: Students will brainstorm and share ideas like automating file operations, simple data analysis, web development, game scripting, scientific computing, etc. Instructor will guide the discussion to highlight Python's versatility.

# **คำตอบ**
 โดยการเขียนสคริปด้วย python นั้น สามารถนำไปใช้ประโยชน์ได้หลายอย่างเช่น ใช้หาค่าพื้นที่ใต้กราฟ ใช้ดึงรายชื่อนักศึกษาหรือคะแนนเพื่อนำมาคำนวณและตัดเกรด ใช้ในการดึงข้อมูลในไฟล์มาใช้เช่นไฟล์ csv ใช้ทำนายค่าด้วย monte carlo

# **Activity 1.2: "Python Puzzlers"**
 - Identifying Data Types & Output
Format: Individual or pair work, followed by a class review.
Instructions: Provide a set of small Python code snippets (on a handout or projected). For each snippet, students must:
Predict the output.
Identify the data type of specific variables after certain operations.
Explain why they chose that output/data type.


In [ ]:
# Puzzler 1
x = 10
y = 3.5
z = x + y
print(z)
print(type(z))

13.5
<class 'float'>


ได้ผลลัพธ์: 13.5 มี type เป็น float

เพราะ x มี type เป็น integer และ y มี type เป็น float เมื่อนำมาบวกกันจะได้เป็น float เนื่องจาก float บวกกับ integer จะได้ float เสมอ

In [ ]:
# Puzzler 2
name = "Alice"
age = "30" # Note: This is a string
message = name + " is " + age + " years old."
print(message)
print(type(age))

Alice is 30 years old.
<class 'str'>


ได้ผลลัพธ์: Alice is 30 years old. มี type เป็น string

เพราะ name มี type เป็น string และ age มี type เป็น string เมื่อนำ string มาบวกกันจะเป็นการต่อข้อความกันแทน ทำให้ message เป็น string ที่เป็นข้อความที่ต่อกัน

In [ ]:
# Puzzler 3
is_sunny = True
is_raining = False
print(is_sunny and is_raining)
print(type(is_sunny))

False
<class 'bool'>


ได้ผลลัพธ์: False มี type เป็น boolean

เพราะ is_sunny เป็น boolean ที่มีค่าเป็น True ส่วน is_raining เป็น boolean ที่มีค่าเป็น False โดยเมื่อนำทั้ง is_sunny และ is_raining มาดำเนินการทางตรรกศาสตร์ ด้วย and โดย and นั้นถ้ามีอย่างน้อย 1 ตัวเป็น False ก็จะได้ผลลัพธ์เป็น False ทันที

In [ ]:
# Puzzler 4
# What happens here?
# num_str = input("Enter a number: ")
# result = num_str + 5
# print(result)

ผลลัพธ์ : ไม่มีผลลัพธ์ เนื่องจากทุกบรรทัดมีเครื่องหมายหมายเหตุ(Comment) อยู่ด้านหน้าทำให้ไม่มีส่วนไหนของโค้ดนี้เกิดการทำงาน